gramps/gen/display/name.py

```diff
@@ -1007,8 +1007,16 @@ def name_grouping_data(self, db, pn):
         """
         if pn[_GROUP]:
             return pn[_GROUP]
-        return db.get_name_group_mapping(_raw_primary_surname_only(
+        name = pn[_GROUP]
```

Comment:
If you are modifying the `name_grouping_data` method you should probably make similar changes to the  `name_grouping_name` method.

Reply:
If I look at where the name_grouping_name method is used, I only see it's called by the name_grouping method.
This last method is never called either in gramps or in addons.
Where could we call this method?
I think the two method name_grouping_name and name_grouping could be removed.